app/routes.py

```diff
@@ -1 +1,280 @@
+            if not by or by == 'title':
+                order = cls.title.asc
+            if by == 'goal':
+                order = cls.goal_id.asc
+            if by == 'date':
+                order = cls.completed_at.asc
+        if sort_query == 'desc':
+            if not by or by == 'title':
+                order = cls.title.desc
+            if by == 'goal':
+                order = cls.goal_id.desc
+            if by == 'date':
+                order = cls.completed_at.desc
+
+        results = cls.query.order_by(order())
+    else:
+        results = cls.query
+
+    if filter_query:
+        if filter_query == 'todo':
+            results = results.filter(Task.completed_at == None)
+        if filter_query == 'done':
+            results = results.filter(Task.completed_at != None)
+
+    return results
+
+#---------------------------------------------------------------#
+#-------------------------TASK ROUTES---------------------------#
+#---------------------------------------------------------------#
+
+#-------------GET ALL TASKS------------#
+@task_bp.route("", methods =["GET"])
+def get_tasks():
+    tasks = query_filter(Task)
+    tasks_response = [to_dict(task) for task in tasks]
+
+    return make_response(jsonify(tasks_response), 200)
+
+#-------------MAKE NEW TASK ------------#
+@task_bp.route("", methods =["POST"])
+def make_task():
+    request_body = request.get_json()
+    new_task = validate_new_item(Task,request_body)
+    db.session.add(new_task)
+    db.session.commit()
+
+    return make_response({"task":to_dict(new_task)}, 201)
+
+#-------------GET SINGLE TASK ------------#
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_item(Task,task_id)
+
+    return make_response(dict(task=to_dict(task)), 200)                  
+
+#-------------EDIT SINGLE TASK ------------#
+@task_bp.route("/<task_id>", methods=["PUT"])
+def edit_one_task(task_id):
+    task = validate_item(Task,task_id)
+    request_body = request.get_json()
+
+    if "title" in request_body:
+        task.title=request_body["title"]
+    if "description" in request_body:
+        task.description=request_body["description"]
+    if "completed_at" in request_body:
+        task.completed_at=request_body["completed_at"]
+
+    db.session.commit()
+
+    return(make_response({"task":to_dict(task)}, 200))
+
+#-------------DELETE SINGLE TASK ------------#
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    task = validate_item(Task,task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response({"details": f'Task {task.task_id} "{task.title}" successfully deleted'},200)
+
+#-------------MARK TASK COMPLETE------------#
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def check_task(task_id):
+    task = validate_item(Task,task_id)
+    task.completed_at=datetime.now()
+
+    db.session.commit()
+
+    post_to_slack(task)
```

Comment:
We might want to wrap this in a try/except since if the call to Slack fails, we will presumably still want to return a success message to the client.